designsafe/apps/auth/tasks.py

```diff
@@ -17,76 +21,68 @@
 logger = logging.getLogger(__name__)
 
 
+# TODOV3 replace following by importing mkdir from server/portal/libs/agave/operations.py
+# TODOV3 when added during https://tacc-main.atlassian.net/browse/DES-2703
+def mkdir(client, system, path, dir_name):
+    client.files.mkdir(systemId=system, path=path_input)
+    return {"result": "OK"}
+
+
 @shared_task(default_retry_delay=30, max_retries=3)
-def check_or_create_agave_home_dir(username, systemId):
+def check_or_configure_system_and_user_directory(username, system_id, create_credentials):
+    path = username
+
     try:
-        # TODO should use OS calls to create directory.
-        logger.info(
-            "Checking home directory for user=%s on "
-            "default storage systemId=%s",
-            username,
-            systemId
+        user_client = get_user_model().objects.get(username=username).tapis_oauth.client
+        user_client.files.listFiles(
+            systemId=system_id, path=path
         )
-        ag = Agave(api_server=settings.AGAVE_TENANT_BASEURL,
-                   token=settings.AGAVE_SUPER_TOKEN)
+        logger.info(f"Checked and there is no need to configure system:{system_id} for {username}")
+        return
+    except ObjectDoesNotExist:
+        logger.info(f"Checked and there is a need to configure system:{system_id} for {username} (new user)")
+        pass
+    except BaseTapyException as e:
+        logger.info(f"Checked and there is a need to configure system:{system_id} for {username}: {e}")
```

Comment:
Great logs to have. Maybe we can come up with a more clear way to communicate it with like a prefix? At a glance it's not easy to distinguish between the three.

Reply:
good idea.  adding prefix.